spark/v4.0/spark/src/main/java/org/apache/iceberg/spark/data/ParquetWithSparkSchemaVisitor.java

```diff
@@ -182,19 +183,17 @@ private static <T> T visitField(
 
   private static <T> List<T> visitFields(
       StructType struct, GroupType group, ParquetWithSparkSchemaVisitor<T> visitor) {
-    StructField[] sFields = struct.fields();
-    Preconditions.checkArgument(
-        sFields.length == group.getFieldCount(), "Structs do not match: %s and %s", struct, group);
     List<T> results = Lists.newArrayListWithExpectedSize(group.getFieldCount());
-    for (int i = 0; i < sFields.length; i += 1) {
-      Type field = group.getFields().get(i);
-      StructField sField = sFields[i];
-      Preconditions.checkArgument(
-          field.getName().equals(AvroSchemaUtil.makeCompatibleName(sField.name())),
-          "Structs do not match: field %s != %s",
-          field.getName(),
-          sField.name());
-      results.add(visitField(sField, field, visitor));
+    for (StructField sField : struct.fields()) {
+      String fieldName = AvroSchemaUtil.makeCompatibleName(sField.name());
```

Comment:
@Fokko @rdblue will using field name be a problem if the field is renamed in Iceberg table? would the mapping be messed up during read or write?

Reply:
Yes, that's correct. The read path doesn't use this.